src/Psl/Xml/Reader/ReaderResourceInterface.php

```diff
@@ -0,0 +1,56 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Psl\Xml\Reader;
+
+use Psl\File\Exception\FileNotFoundException;
+use XMLReader;
+
+/**
+ * This resource is bugging me.
+ * There is no way to pass a 'resource' into the XMLReader at the moment.
+ *
+ * Therefore, it is not possible to use a IO ReadHandle
+ * unless we do some bad stuff like streaming it to a temp file and read that file.
+ *
+ * This resource interface could be removed if we added these named constructors on the reader interface.
+ * Both are not optimal .. but neither is life!
+ *
+ * @see https://www.php.net/manual/en/xmlreader.open.php
+ * @see https://www.php.net/manual/en/xmlreader.xml.php
+ */
```

Comment:
FYI : The same goes for DOMDocument and simplexml : they can only load a complete XML string or a URI.